### PR TITLE
feat: add Google Sign in support for Web platform

### PR DESCRIPTION
**Pull request by ferrarafer**
- Added AdditionalUserInfo to FirebaseAuthenticationResult, has some data that may be useful
- Added Web support to `GoogleSignIn`
- Disabled test coverage

### CODE REVIEW

**Review by FilledStacks — CHANGES_REQUESTED** [review]
Awesome. 

@ferrarafer some changes based on the new workflow. 

- Version will automatically be updated
- The changelog will be created using your commit message (feat)

Remove the version update lets see if @buehler 's setup works 😄

**Comment by buehler** [comment]
Hey @FilledStacks @ferrarafer 
I hope it will, just keep in mind that the commit message must adhere to angular commit convention:

e.g.:
`feat: add Google Sign in support for web platform`

**Comment by buehler** [comment]
(you don't need to update version number and changelog file)

**Comment by buehler** [comment]
and you may need to fix linting / test errors from the old code (did not do that for all repos).

you can do that by using:
`dart fix --apply` and `dart format .`

**Comment by ferrarafer** [comment]
> and you may need to fix linting / test errors from the old code (did not do that for all repos).
> 
> you can do that by using: `dart fix --apply` and `dart format .`

Done ✅

**Comment by ferrarafer** [comment]
> Awesome.
> 
> @ferrarafer some changes based on the new workflow.
> 
> * Version will automatically be updated
> * The changelog will be created using your commit message (feat)
> 
> Remove the version update lets see if @buehler 's setup works 😄

Done ✅ ... let's see if it works 😄

**Comment by ferrarafer** [comment]
@buehler Great work, thanks for the effort ❤️

**Comment by FilledStacks** [comment]
@ferrarafer can you remove the code coverage line from the workflow. We don't have tests in this package.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
       - run: flutter pub get
       - run: dart format --fix --set-exit-if-changed .
       - run: flutter analyze
-      - run: flutter test --coverage
+      # - run: flutter test --coverage
```
Comment (buehler) [inline .github/workflows/test.yml:23]: comment out everything below this line, then the action should work

Comment (FilledStacks) [reply .github/workflows/test.yml:23]: @buehler I removed you from the organization but didn't know how to add you back as a collaborator 😅  I don't quite know how the seats work. It says $4 per seat, but it' shows 0 cost and you said it's free as well.

**Review by buehler — APPROVED** [review]

**Comment by ferrarafer** [comment]
I'm ready to see the show 😎

**Comment by buehler** [comment]
me 2 🙈

**Review by FilledStacks — APPROVED** [review]
Ready to gooooooooooo

**Comment by buehler** [comment]
Wow. Color me impressed. It really worked 🎉

**Comment by ferrarafer** [comment]
Is fine 2.19.0 instead of 0.2.19? 🤔

**Comment by buehler** [comment]
Haha, you know, those systems normally fail 2 or 3 times before they start working 😂 
@ferrarafer well.... you found an error. it should've been 0.3.0 (since its a feature)

**Comment by buehler** [comment]
can we pull the package from pub.dev? as far as I know, publishing is kind of forever. We can fix that the next number is 0.3.0

**Comment by buehler** [comment]
otherwise: it can stay on 2.19 (which is bad, but it's just a number... your call)

**Comment by ferrarafer** [comment]
That's a @FilledStacks decision 😅

**Comment by buehler** [comment]
if you can delete (maybe inside a certain timeframe) the package from pub.dev, we can fix it perfectly. if not: we just need to make sure when the time of 2.19 arrives (in the future) that we skip that version :)

**Comment by buehler** [comment]
ok, you could:
```
However, because mistakes can occasionally happen, exceptions can be granted to this general policy at the pub.dev moderators’ discretion, on email request to support@pub.dev. Examples of circumstances under which exceptions may be granted include the following:

A package was published by mistake, and the publisher files an unpublish request within 48 hours. The pub.dev moderators will ensure that no widespread usage of the package has begun, and if so, may unpublish the package.
A package violates the Naming, Content, or Copyright policies (see below), and a user files a moderation request.
The package violates the [Google Terms of Service](https://policies.google.com/terms?hl=en) or other applicable terms of service.
```

So we may be able to delete that version and I'll create a tag that will deploy v0.3.0 (correctly)

**Comment by buehler** [comment]
(I'm not an admin anymore :) i cannot delete the tag and stuff...)
so:
- delete the two tags (2.18 and 2.19) and the release
- manually create a 0.2.18 tag and push it
- create an empty commit with some feature message (i.e. this commit message from this PR)
- push that into main -> the new 0.3.0 should be created and released

**Comment by ferrarafer** [comment]
@buehler Can we recap?

1. delete the following tags: `2.18.0`, `2.19.0`, `release`
2. create `0.2.18` tag on the latest commit before this PR and push it
3. create an empty commit with the following message (I still need to find how to create a commit without changes 😅 )
    - feat: add Google Sign in support for Web platform
4.  push to main

**Comment by buehler** [comment]
Hey @ferrarafer 
Happy to clarify (I was traveling this weekend..):
since the release has been more than 48h I don't think we can pull it off pub.dev anymore. So we would need to stick with that version (or we don't but then we need to pay attention to skip the unwanted version numbers).

if so:
1. Delete the git release and git tag for the unwanted version(s) (i.e. `2.18` and `2.19`)
2. create the `0.2.18` as you wrote
3. create the empty (on latest) commit with: `git commit --allow-empty -m "feat: add Google Sign in support for Web platform"` and push it

It is a new feature, which increases the minor version of a repository. Fixes will increase the patch version and breaking changes the major version. You can read about it here: https://www.conventionalcommits.org/en/v1.0.0-beta.4/ and https://semver.org/lang/de/

**Comment by buehler** [comment]
Hey @ferrarafer, directly into the main is impossible when you're not an admin. (Branch protection rule)

**Pull request merged**